pennylane/noise/conditionals.py

```diff
@@ -0,0 +1,481 @@
+    """Computes the class for a given argument from its string name, instance,
+    or a sequence of them.
+
+    Args:
+        val (Union[str, Operation, Iterable]): object to be used
+            for building the operation set.
+
+    Returns:
+        tuple[class]: tuple of :class:`Operation <pennylane.operation.Operation>`
+        classes corresponding to val.
+    """
+    vals = val if isinstance(val, (list, tuple, set)) else [val]
+    return tuple(
+        (
+            getattr(qml.ops, val, None) or getattr(qml, val)
+            if isinstance(val, str)
+            else (val if isclass(val) else getattr(val, "__class__"))
+        )
+        for val in vals
+    )
+
+
+def _check_arithmetic_ops(op1, op2):
+    """Helper method for comparing two arithmetic operators based on type check of the bases"""
+    # pylint: disable = unnecessary-lambda-assignment
+
+    if isinstance(op1, (Controlled, ControlledSequence)) or isinstance(
+        op2, (Controlled, ControlledSequence)
+    ):
+        return (
+            isinstance(op1, type(op2))
+            and op1.arithmetic_depth == op2.arithmetic_depth
+            and _get_ops(op1.base) == _get_ops(op2.base)
+        )
+
+    lc_cop = lambda op: qml.ops.LinearCombination(*op.terms())
+
+    if isinstance(op1, qml.ops.Exp) or isinstance(op2, qml.ops.Exp):
+        if (
+            not isinstance(op1, type(op2))
+            or (op1.base.arithmetic_depth != op2.base.arithmetic_depth)
+            or not qml.math.allclose(op1.coeff, op2.coeff)
+            or (op1.num_steps != op2.num_steps)
+        ):
+            return False
+        if op1.base.arithmetic_depth:
+            return _check_arithmetic_ops(op1.base, op2.base)
+        return _get_ops(op1.base) == _get_ops(op2.base)
+
+    op1, op2 = qml.simplify(op1), qml.simplify(op2)
+    if op1.arithmetic_depth != op2.arithmetic_depth:
+        return False
+
+    coeffs, op_terms = lc_cop(op1).terms()
+    sprods = [_get_ops(getattr(op_term, "operands", op_term)) for op_term in op_terms]
+
+    def _lc_op(x):
+        coeffs2, op_terms2 = lc_cop(x).terms()
+        sprods2 = [_get_ops(getattr(op_term, "operands", op_term)) for op_term in op_terms2]
+        for coeff, sprod in zip(coeffs2, sprods2):
+            present, p_index = False, -1
+            while sprod in sprods[p_index + 1 :]:
+                p_index = sprods[p_index + 1 :].index(sprod) + (p_index + 1)
+                if qml.math.allclose(coeff, coeffs[p_index]):
+                    coeffs.pop(p_index)
+                    sprods.pop(p_index)
+                    present = True
+                    break
+
+            if not present:
+                break
+
+        return present
+
+    return _lc_op(op2)
+
+
+def op_in(ops):
+    """Builds a conditional as a boolean function for evaluating
+    if a given operation exist in a specified set of operations.
+
+    Args:
+        ops (str, class, Operation, Union(list[str, class, Operation])): sequence of string
```

Comment:
```suggestion
        ops (str, class, Operation, list(Union[str, class, Operation])): sequence of string
```
Should that be the order?